### PR TITLE
Change fee estimate multiplier

### LINKED ISSUE #702: Investigate max_fee not high enough when `auto_estimate`
### Feature Request

Perhaps a multiplier should be increased from `1.1` that it is currently.

### PR DESCRIPTION
**Pull request by drknzz**
Closes #702

## Introduced changes

- Increase automatic estimated fee multiplier from `1.1` to `1.5`

### CODE REVIEW

File: starknet_py/net/account/account_client.py [hunk 1]
```diff
@@ -326,7 +326,7 @@ async def _get_max_fee(
 
         if auto_estimate:
             estimate_fee = await self.estimate_fee(transaction)
-            max_fee = int(estimate_fee.overall_fee * 1.1)
+            max_fee = int(estimate_fee.overall_fee * ESTIMATED_FEE_MULTIPLIER)
```
Comment (cptartur) [inline starknet_py/net/account/account_client.py:329]: I wouldn't change it in AccountClient: it is deprecated anyways.

File: starknet_py/net/account/account.py [hunk 2]
```diff
@@ -110,7 +111,7 @@ async def _get_max_fee(
 
         if auto_estimate:
             estimate_fee = await self._estimate_fee(transaction)
-            max_fee = int(estimate_fee.overall_fee * 1.1)
+            max_fee = int(estimate_fee.overall_fee * ESTIMATED_FEE_MULTIPLIER)
```
Comment (cptartur) [inline starknet_py/net/account/account.py:114]: I think we could make `fee_multiplier` a class variable, but don't expose it in constructor. This way `Account` would use default of `1.5` but could adjusted by the user like `account.fee_multiplier = 2.0`.

cc @war-in

**Review by cptartur — CHANGES_REQUESTED** [review]

File: starknet_py/net/account/account.py (hunk 2 above)
Comment (war-in) [reply starknet_py/net/account/account.py:114]: Yes, I would like to have it too

File: starknet_py/net/account/account.py [hunk 3]
```diff
@@ -55,6 +55,8 @@ class Account(BaseAccount):
     Default Account implementation.
     """
 
+    ESTIMATED_FEE_MULTIPLIER: float = 1.5
```
Comment (cptartur) [inline starknet_py/net/account/account.py:58]: I wouldn't use uppercase here and perhaps add a docstring.

File: docs/guide/using_existing_contracts.rst [hunk 4]
```diff
@@ -61,7 +61,7 @@ Automatic fee estimation
 
 For testing purposes it is possible to enable automatic fee estimation when making
 a transaction. Starknet.py will then use ``estimate_fee()`` internally and use value
-returned by it multiplied by ``1.1`` as a ``max_fee``.
+returned by it multiplied by ``1.5`` as a ``max_fee``.
```
Comment (cptartur) [inline docs/guide/using_existing_contracts.rst:64]: Would add information about adjusting the default multiplier here or somewhere else.

File: starknet_py/net/account/account.py (hunk 3 above)
Comment (drknzz) [reply starknet_py/net/account/account.py:58]: https://stackoverflow.com/a/45702782

Comment (cptartur) [reply starknet_py/net/account/account.py:58]: But is it really a constant if it is intended to be changed by the user?

File: docs/guide/using_existing_contracts.rst (hunk 4 above)
Comment (drknzz) [reply docs/guide/using_existing_contracts.rst:64]: Adjusting? You mean the fact that it has been changed or? It cannot be adjusted manually

File: starknet_py/net/account/account.py (hunk 3 above)
Comment (drknzz) [reply starknet_py/net/account/account.py:58]: It's not changeable yet though

File: docs/guide/using_existing_contracts.rst (hunk 4 above)
Comment (cptartur) [reply docs/guide/using_existing_contracts.rst:64]: I'd like it to be adjustable by the user. That's why I wanted the  `ESTIMATED_FEE_MULTIPLIER` to be a part of Account now.

File: starknet_py/net/account/account.py (hunk 3 above)
Comment (drknzz) [reply starknet_py/net/account/account.py:58]: Well it is if one sets account.attr = 2.0

**Review by war-in — APPROVED** [review]

File: starknet_py/net/account/account.py [hunk 5]
```diff
@@ -53,8 +53,12 @@
 class Account(BaseAccount):
     """
     Default Account implementation.
+
+    :ivar ESTIMATED_FEE_MULTIPLIER: Amount by which each estimated fee is multiplied when using `auto_estimate`.
     """
 
+    ESTIMATED_FEE_MULTIPLIER: float = 1.5
```
Comment (cptartur) [inline starknet_py/net/account/account.py:60]: Wouldn't the docs for this generate correctly if you just added a docstring directly below `ESTIMATED_FEE_MULTIPLIER`, without the `:ivar ESTIMATED_FEE_MULTIPLIER:`?

That's how it worked with dataclasses.

File: docs/guide/using_existing_contracts.rst [hunk 6]
```diff
@@ -71,3 +71,8 @@ returned by it multiplied by ``1.1`` as a ``max_fee``.
 .. code-block:: python
 
     await contract.functions["put"].invoke(k, v, auto_estimate=True)
+
+
+.. note::
+    It is possible to set the value by which the estimated fee is multiplied,
```
Comment (cptartur) [inline docs/guide/using_existing_contracts.rst:77]:
```suggestion
    It is possible to configure the value by which the estimated fee is multiplied,
```

**Review by cptartur — APPROVED** [review]

File: starknet_py/net/account/account.py (hunk 5 above)
Comment (drknzz) [reply starknet_py/net/account/account.py:60]: I think it wouldn't show at all in the class

Comment (cptartur) [reply starknet_py/net/account/account.py:60]:
```py
ESTIMATED_FEE_MULTIPLIER: float = 1.5
"""Amount by which each estimated fee is multiplied when using `auto_estimate`."""
```
should work and display as 
image

**Pull request merged**